pkg/operator/controller/vulnerabilityreport.go

```diff
@@ -228,8 +230,17 @@ func (r *VulnerabilityReportReconciler) submitScanJob(ctx context.Context, owner
 		WithTolerations(scanJobTolerations).
 		WithAnnotations(scanJobAnnotations).
 		WithCredentials(credentials).
+		WithClient(r.Client).
 		Get()
 
+	if err != nil {
+		if err.Error() == "cannot get nodename from podspec" {
```

Comment:
We should either use global variable to do such comparisons (e.g. kube.ErrUnknownNodeName) or return blank string when we resolve node name. Anyway, we should probably move this logic away and retrieve node name in Trivy plugin impl.